### PR TITLE
Enable upgrade to next supported Kubernetes version

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**: Remove constraint of only being able to upgrade to the next minor version if the next minor version is deprecated. Instead, allow upgrading to the next minor version OR the next available minor version. So if I have 1.7 cluster and 1.7 and 1.8 are deprecated, I can upgrade to 1.9 but not 1.10+. If I have a 1.9 cluster I can upgrade to 1.9 or 1.10.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
https://jenkins.azure-containers.io/view/acs-engine%20ad-hoc/job/k8s-upgrade/176/console

File: pkg/api/orchestrators_test.go [hunk 1]
```diff
@@ -298,3 +238,44 @@ func TestDockerceInfoInfo(t *testing.T) {
 	_, e := dockerceInfo(csOrch, false)
 	Expect(e).To(BeNil())
 }
+
+func TestGetKubernetesAvailableUpgradeVersions(t *testing.T) {
```
Comment (CecileRobertMichon) [inline pkg/api/orchestrators_test.go]: This is one of those "the tests are just as important if not more important than the code" PRs

**Comment by CecileRobertMichon** [comment]
/hold 

TODO: retest upgrade and back compat

**Comment by CecileRobertMichon** [comment]
https://jenkins.azure-containers.io/view/acs-engine%20ad-hoc/job/k8s-upgrade/177/console

**Comment by CecileRobertMichon** [comment]
/unhold

back compat upgrade passed

**Comment by tariq1890** [comment]
@CecileRobertMichon I think we should leverage the capabilities of semver library. I have this link for reference - https://github.com/Masterminds/semver/blob/master/README.md#sorting-semantic-versions.

My suggestion is that we use sort and then use the sorted array to get the Max and Min versions. This will simplify the logic by a great deal and do away with the ifs and for

**Comment by tariq1890** [comment]
Based on the `pre-release` bool, we can choose to include the pre-release versions when in the array to be sorted.

**Pull request merged**